docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
+.NET Core SDK=3.0.100-preview-009812
+  [Host]     : .NET Core 3.0.0-preview-27122-01 (CoreCLR 4.6.27121.03, CoreFX 4.7.18.57103), 64bit RyuJIT
+  Job-SBBRNM : .NET Core ? (CoreCLR 4.6.27416.73, CoreFX 4.7.19.11901), 64bit RyuJIT
+```
+
+As you can see, before the "update" the version for the Job (not the Host process) was `4.6.27416.73`.
+
+The CoreFX build system has a built-in support for that, exposed by the [CoreCLROverridePath](https://github.com/dotnet/corefx/blob/0e7236fda21a07302b14030c82f79bb981c723a6/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits) build parameter:
+
+```cmd
+C:\Projects\corefx> build -Release /p:CoreCLROverridePath="C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release"
+```
+
+After the update, the reported CoreCLR version should be changed:
+
+```log
+  [Host]     : .NET Core 3.0.0-preview-27122-01 (CoreCLR 4.6.27121.03, CoreFX 4.7.18.57103), 64bit RyuJIT
+  Job-CYVJFZ : .NET Core ? (CoreCLR 4.6.27517.0, CoreFX 4.7.19.11901), 64bit RyuJIT
+```
+
+**It's very important to validate the reported version numbers** to make sure that you are running the benchmarks using the versions you think that you are using.
+
+## Benchmarking new API
+
+When developing new CoreFX features, we should be thinking about the performance from day one. One part of doing this is writing benchmarks at the same time when we write our first unit tests. Keeping the benchmarks in a separate repository makes it a little bit harder to run the benchmarks against new CoreFX API, but it's still very easy.
+
+### Reference
+
+When you develop a new feature, whether it's a new method/type/library in CoreFX all you need to do is to build it in Release and just reference the produced `.dll` from the [MicroBenchmarks.csproj](../src/benchmarks/micro/MicroBenchmarks.csproj) project file.
+
+The easiest way to do it is to open [MicroBenchmarks.sln](../src/benchmarks/micro/MicroBenchmarks.sln) with Visual Studio, right click on the [MicroBenchmarks](../src/benchmarks/micro/MicroBenchmarks.csproj) project file, select "Add", then "Reference..." and in the new Dialog Window click "Browse" in the left bottom corner. From the File Picker, choose the new library and click "Add". Please don't forget to Save the changes (Ctrl+Shift+S). From this moment you should be able to consume new public types and methods exposed by the referenced library.
+
+Sample changes:
+
+```cs
+namespace System
+{
+    public static class Console
+    {
+        public static void WriteHelloWorld() => WriteLine("Hello World!");
+        
+        // the rest omitted for brevity
+    }
+}
+```
+
+Sample project file change:
+
+```xml
+<ItemGroup>
+  <Reference Include="System.NewAPI">
+    <HintPath>..\..\..\..\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\System.Console.dll</HintPath>
+  </Reference>
+</ItemGroup>
+```
+
```

Comment:
AFAIK if new api is on coreclr and after exposed to corefx we can only test using custom CoreRun local coreclr+corefx build and not with python or simple `dotnet run -f netcoreapp3.0 --filter System.Memory*`.
If so maybe we could specify this.

Reply:
Most of the CoreFX contributors have dotnet cli installed, so I have based this workflow doc on using dotnet cli (not the python script). 

But when it comes to CoreCLR contributors I think that having an up-to-date dotnet cli is rare, so the workflow for them is going to be based on the python script. I am going to mention it there.